pkg/segment/structs/evaluationstructs.go

```diff
@@ -1607,13 +1616,56 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 				return 0, fmt.Errorf("NumericExpr.Evaluate: cannot convert '%v' to number with base %d", strValue, base)
 			}
 			return float64(number), nil
+		case "relative_time":
+			if self.Left == nil {
+				return 0, fmt.Errorf("NumericExpr.Evaluate: relative_time operation requires a non-nil left operand")
+			}
+
+			var epochTime int64
+			var err error
+			// 1. Direct epoch input (e.g., relative_time(1234567890, "-1d")) is stored in self.Value.
+			// 2. Epoch from another command (e.g., relative_time(now(), "-1d")) after evaluation is stored in left, leaving self.Value as an empty string.
+			if self.Value != "" {
+				epochTime, err = strconv.ParseInt(self.Value, 10, 64)
+				if err != nil {
+					return 0, fmt.Errorf("NumericExpr.Evaluate: failed to parse epoch time from Value %v, err: %v", self.Value, err)
+				}
+			} else if left >= 0 {
+				epochTime = int64(left)
```

Comment:
Would it make sense to always store it in the left so we don't have to have this logic. Or will that complicate things so doing it this way is better?

Reply:
The value stored in `left` comes from nested commands like `now()` or `time()`, and the current parsing of the `relative_time` command does not handle this. Do you recommend that we also store the parsed time from the `relative_time` command in `left` in this case? I thought it might be better if we store the evaluated value from a command like `now()` or `time()` separately from the passed value in the query, such as epoch `1234567890`. If we update it to be stored in `left`, we would also have to test that an evaluated command used in the query in a different way does not override the parsed value from `relative_time` command.

Reply:
I'm not sure why we'd want to store an exact value like `1234567890` in a different place from something like `now()` that gets evaluated to `1234567890`. Why do you think it would be good to store those separately? To me it seems like we'd want to store them both similarly so we could do something like `self.Left.Evaluate()` and get `1234567890`, whether that was a direct value specified by the user or the result of running `now()`

Also I don't understand what you're saying here:
>  If we update it to be stored in left, we would also have to test that an evaluated command used in the query in a different way does not override the parsed value from relative_time command.

Reply:
Andrew, it looks like both exact value and evaluated values were stored in `left`. `self.Value` would have the exact value only, so we can ignore the `self.Value` and just use the values from `left`, since `left` would have the value irrespective of whether it is an exact value or an evaluated value. I have updated the logic to use only the value from `left` in all cases and added test case using `now()` and `time()` as well.